outputs/policyreport.go

```diff
@@ -0,0 +1,321 @@
+		if err != nil {
+			log.Printf("[ERROR] : PolicyReport - Unable to load kube config file: %v\n", err)
+		}
+	}
+	crdclient, err := crdClient.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		OutputType:      "PolicyReport",
+		Config:          config,
+		Stats:           stats,
+		PromStats:       promStats,
+		StatsdClient:    statsdClient,
+		DogstatsdClient: dogstatsdClient,
+		Crdclient:       crdclient,
+	}, nil
+}
+
+// UpdateOrCreatePolicyReport creates/updates PolicyReport/ClusterPolicyReport Resource in Kubernetes
+func (c *Client) UpdateOrCreatePolicyReport(falcopayload types.FalcoPayload) {
+	c.Stats.PolicyReport.Add(Total, 1)
+
+	event, namespace := newResult(falcopayload)
+
+	var err error
+	if namespace != "" {
+		// case where the event is namespace specific
+		err = updatePolicyReports(c, namespace, event)
+	} else {
+		err = updateClusterPolicyReport(c, event)
+	}
+	if err == nil {
+		go c.CountMetric(Outputs, 1, []string{"output:policyreport", "status:ok"})
+		c.Stats.PolicyReport.Add(OK, 1)
+		c.PromStats.Outputs.With(map[string]string{"destination": "policyreport", "status": OK}).Inc()
+	} else {
+		go c.CountMetric(Outputs, 1, []string{"output:policyreport", "status:error"})
+		c.Stats.PolicyReport.Add(Error, 1)
+		c.PromStats.Outputs.With(map[string]string{"destination": "policyreport", "status": Error}).Inc()
+	}
+}
+
+//newResult creates a new entry for Reports
+func newResult(FalcoPayload types.FalcoPayload) (_ *wgpolicy.PolicyReportResult, namespace string) {
+	var properties = make(map[string]string)
+	for property, value := range FalcoPayload.OutputFields {
+		if property == "ka.target.namespace" || property == "k8s.ns.name" {
+			namespace = fmt.Sprintf("%v", value) // not empty for policy reports
+		}
+		properties[property] = fmt.Sprintf("%v", value)
+	}
+	if FalcoPayload.Priority > types.Priority(minimumPriority) {
+		severity = highpriority
+	} else if FalcoPayload.Priority < types.Priority(minimumPriority) {
+		severity = lowpriority
+	} else {
+		severity = mediumpriority
+	}
+
+	return &wgpolicy.PolicyReportResult{
+		Policy:      FalcoPayload.Rule,
+		Category:    "SI - System and Information Integrity",
+		Source:      policyReportSource,
+		Scored:      false,
+		Timestamp:   metav1.Timestamp{Seconds: int64(FalcoPayload.Time.Second()), Nanos: int32(FalcoPayload.Time.Nanosecond())},
+		Severity:    v1alpha2.PolicyResultSeverity(severity),
+		Result:      "fail",
+		Description: FalcoPayload.Output,
+		Properties:  properties,
+	}, namespace
+}
+
+//check for low priority events to delete first
+func checklow(result []*wgpolicy.PolicyReportResult) (swapint int) {
+	for i, j := range result {
+		if j.Severity == mediumpriority || j.Severity == lowpriority {
+			return i
+		}
+	}
+	return -1
+}
+
+//update summary for clusterpolicyreport 'report'
+func updateClusterPolicyReportSummary(event *wgpolicy.PolicyReportResult) {
+	if event.Severity == highpriority {
```

Comment:
This should not depend on the severity. It should depend on the Result (fail / warn / pass / etc.)

Reply:
Hey @fjogeleit , the summary(fail/warn) depends on the severity of the event being added. The severity is decided by the user, the user could choose the minimum priority above which they want events to be mapped to fail. In this block we are updating the summary based on the event in question, if it is of high severity (fail) it is mapped to fail Summary else it is mapped to warn.

Reply:
As you can read on the spec https://htmlpreview.github.io/?https://github.com/kubernetes-sigs/wg-policy-prototypes/blob/master/policy-report/docs/index.html#wgpolicyk8s.io/v1alpha2.PolicyReportSummary summary is a count on results by ResultStatus. It is Independed from severity.